intentgen/src/main/java/com/chatbot/IntentGenerator.java

```diff
@@ -0,0 +1,190 @@
+    final UseCase.Builder useCaseBuilder = UseCase.newBuilder();
+    // the use case file is a proto text file, so it is parsed using the textformat parser
+    final TextFormat.Parser textFromatParser = TextFormat.getParser();
+    textFromatParser.merge(fileContent, useCaseBuilder);
+    return useCaseBuilder.build().getIntentsList();
+  }
+
+  private static Intent.Builder addTrainingPhrases(UseCase.Intent intent,
+      Intent.Builder intentProtobufBuilder) {
+    List<TrainingPhrase> trainingPhrasesProtobufList = new ArrayList<TrainingPhrase>();
+    List<String> trainingPhrasesStringList =
+        new ArrayList<String>(intent.getTrainingPhrasesList());
+    if(intent.getIntentType() == UseCase.IntentType.YES) {
+      trainingPhrasesStringList.addAll(YES_TRAINING_PHRASES);
+    }
+    else if (intent.getIntentType() == UseCase.IntentType.NO) {
+      trainingPhrasesStringList.addAll(NO_TRAINING_PHRASES);
+    }
+    List<TrainingPhrase> userTrainingPhrases = trainingPhrasesStringList.stream()
+        .map(trainingPhrase -> encodedStringToTrainingPhrase(trainingPhrase))
+        .collect(Collectors.toList()); 
+    trainingPhrasesProtobufList.addAll(userTrainingPhrases);
+    intentProtobufBuilder.addAllTrainingPhrases(trainingPhrasesProtobufList);
+    return intentProtobufBuilder;
+  }
+
+  private static Intent.Builder addResponses(UseCase.Intent intent,
+      Intent.Builder intentProtobufBuilder) {
+    final Intent.Message message = Intent.Message.newBuilder()
+        .setText(Intent.Message.Text.newBuilder().addAllText(intent.getResponsesList()).build())
+        .build();
+    return intentProtobufBuilder.addMessages(message);
+  }
+
+  private static Intent.Builder addOutputContexts(UseCase.Intent intent,
+      Intent.Builder intentProtobufBuilder, String projectID) {
+    List<Context> outputContextProtobufList= new ArrayList<Context>();
+    for(String successor: intent.getSuccessorsList()) {
+      Context contextProtobuf = Context.newBuilder()
+      .setLifespanCount(1)
+      .setName("projects/" + projectID + "/agent/sessions/-/contexts/" + successor + "Context")
+      .build();
+      outputContextProtobufList.add(contextProtobuf);
+    }
+    return intentProtobufBuilder.addAllOutputContexts(outputContextProtobufList);
+  }
+
+  private static void createOrUpdateIntents(List<Intent> intentProtobufList,
+    IntentsClient intentsClient, Map<String, String> intentDisplayNameToName,
+    ProjectAgentName parent) {
+    for(Intent intent: intentProtobufList) {
+      if(intentDisplayNameToName.containsKey(intent.getDisplayName())) {
+        intentsClient.updateIntent(intent, "en");
+      }
+      else {
+        intentsClient.createIntent(parent, intent);
+      }
+    }
+  }
+
+  private static Intent intentToIntentProtobuf(UseCase.Intent intent,
+    Map<String, String> intentDisplayNameToName, String projectID) {
+    Intent.Builder intentProtobufBuilder = Intent.newBuilder();
```

Comment:
do we want to pass the map,projectid every iteration?  could make them class member and this foo non static to use those. Also foo name could be more descriptive like createIntentProtobufForIntent.

Reply:
updated